fuzzywuzzy/process.py

```diff
@@ -216,10 +226,61 @@ def extractOne(query, choices, processor=default_processor, scorer=default_score
         was found that was above score_cutoff. Otherwise, returns None.
     """
     best_list = extractWithoutOrder(query, choices, processor, scorer, score_cutoff)
-    try:
-        return max(best_list, key=lambda i: i[1])
-    except ValueError:
-        return None
+    if char_sort is False:
+        try:
+            return max(best_list, key=lambda i: i[1])
+        except ValueError:
+            return None
+    else:
+        best_list = sorted(best_list, key=lambda i: i[1], reverse=True)
+        try:
+            return max(sortByCommonLetter(best_list, query), key=lambda i: i[1])
+        except ValueError:
+            return None
+
+
+def sortByCommonLetter(sl, query):
+    """This function further sorts the strings with the same scores by common letter count to the query."""
+    current_score, last_index = -1, -1
+    # Iterate over list and look for words with the same scores
+    for i in range(0, len(sl)):
+        # Identify the indexes of the strings with the same scores
+        if sl[i][1] != current_score or i == len(sl) - 1:
+            current_score = sl[i][1]
+            # First iteration, there are no previous words so we do not have to do anything
+            if last_index == -1:
+                last_index = i
+                continue
+            # Found a group of words with the same scores! Now sort them
+            if i - last_index > 1:
+                count_list = []
+                for j in range(last_index, i):
+                    count_list.append((sl[j][0], calculateCommonLetter(query, sl[j][0])))
+                count_list = sorted(count_list, key=lambda k: k[1], reverse=True)
+                # Copy the sorted portion
+                for j in range(0, len(count_list)):
+                    sl[last_index + j] = (count_list[j][0], current_score)
+            last_index = i
+    return sl
```

Comment:
This algorithm is broken for quite a few cases. A couple quick examples:
- When a processor is used the results might not be correct:
```
query_1 = 'COMPANY'
choices_1 = ['Company 1', 'company', 'company', 'Company 2', 'Awesome Company']
process.extractOne(query_1, choices_1, scorer=fuzz.partial_token_set_ratio, char_sort=True)
-> ('Company 1', 100)
```
- it matters wether a element is query or choice
```
query_1 = 'Company'
choices_1 = ['Company 1', 'Company', 'Company 2', 'Awesome Company']
process.extractOne(query_1, choices_1, scorer=fuzz.partial_token_set_ratio, char_sort=True)
-> ('Company 1', 100)
```